lib/addons/src/hooks.ts

```diff
@@ -176,7 +176,7 @@ export const applyHooks = (
     const { hooks } = context;
     hooks.prevMountedDecorators = hooks.mountedDecorators;
     hooks.mountedDecorators = new Set([getStory, ...decorators]);
-    hooks.currentContext = context;
+    hooks.currentContext = Object.assign({}, context, { hooks: null });
```

Comment:
This would ensure the ref to hooks.currentContext is always new/different.

What about:
```suggestion
hooks.currentContext = context || { hooks: null };
```

?